linavigation-base/src/main/java/com/mapbox/navigation/base/trip/TripSession.kt

```diff
@@ -0,0 +1,32 @@
+package com.mapbox.navigation.base.trip
+
+import android.location.Location
+import com.mapbox.android.core.location.LocationEngine
+import com.mapbox.android.core.location.LocationEngineRequest
+import com.mapbox.navigation.base.route.Route
+
+interface TripSession {
+
+    val tripService: TripService
+    val locationEngine: LocationEngine
+    val locationEngineRequest: LocationEngineRequest
+    var route: Route?
+
+    fun startSession()
```

Comment:
> does Session in startSession look redundant?

True, I think it can be simplified.

> what do you think about forcing LocationObserver registration when session is started

IMO it would force the user to listen, while they might not want to yet. When the session is started we'll start requesting updates from the framework already anyway. Having a separate setter for the observer looks more flexible to me.

Reply:
fair. my concern was mostly about that it's not clear that we've started location engine as part of this call and user may not know about it, with callback it's explicit and hard to misuse.